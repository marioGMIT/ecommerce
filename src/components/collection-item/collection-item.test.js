import React from 'react';
import { shallow } from 'enzyme';

import {CollectionItem} from './collection-item.component'
import CustomButton from '../custom-button/custom-button.component';
import { addItem } from '../../redux/cart/cart.actions';

// import { CheckoutItem } from './checkout-item.component';

describe('CollectionItem component', () => {
  let wrapper;
  let mockAddItem;

  beforeEach(() => {
    mockAddItem = jest.fn();    

    const mockProps = {
      item: {
        imageUrl: 'www.testImage.com',
        price: 10,
        name: 'hats',
        quantity: 2
      },
      addItem: mockAddItem,
    };

    wrapper = shallow(<CollectionItem {...mockProps} />);
  });

  it('should render CheckoutItem component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should add an item when add to cart is clicked', () => {
    wrapper
      .find('CustomButton')
      .simulate('click');

    expect(mockAddItem).toHaveBeenCalled();
  });

});