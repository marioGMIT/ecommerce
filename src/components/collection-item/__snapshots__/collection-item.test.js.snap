// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component should render CheckoutItem component 1`] = `ShallowWrapper {}`;
